android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModel.kt

```diff
@@ -78,6 +80,24 @@ constructor(
   private val listResourceDataStateMap =
     mutableStateMapOf<String, SnapshotStateList<ResourceData>>()
 
+  fun decodeBinaryResourceIconsToBitmap(profileId: String) {
+    val profileConfig =
+      configurationRegistry.retrieveConfiguration<ProfileConfiguration>(
+        configId = profileId,
+        configType = ConfigType.Profile
+      )
+    profileConfig.overFlowMenuItems
+      .filter { it.icon != null && !it.icon!!.reference.isNullOrEmpty() }
+      .forEach {
+        val resourceId = it.icon!!.reference!!.extractLogicalIdUuid()
+        viewModelScope.launch(dispatcherProvider.io()) {
+          registerRepository.loadResource<Binary>(resourceId)?.let { binary ->
+            it.icon!!.decodedBitmap = binary.data.decodeToBitmap()
+          }
+        }
+      }
+  }
```

Comment:
Let's add documentation that the icons can only be stored and read from Binary Resources